### ISSUE TITLE
Crash on Android if minimumDate is set

### ISSUE THREAD

**Issue by soemarko** [henninghall/react-native-date-picker#763]
**Describe the bug**
As titled. It was working fine prior to updating to the latest version.

**Expected behavior**
Not crash

**To Reproduce**
Add example code that reproduces the behavior. 
```javascript
export default function App() {

  const [date, setDate] = useState(new Date());

  return (
    <DatePicker
              date={date}
              onDateChange={setDate}
              androidVariant='nativeAndroid'
              minimumDate={new Date()}
            />
  );
}
```

If I commented out `minimumDate`, it works. I think this has something to do with the latest update.

**Smartphone (please complete the following information):**
 - OS: Android Simulator
 - React Native version 0.71.1
 - react-native-date-picker version 4.3.6

**Crash Screenshot**

<img width="328" alt="Screenshot 2024-02-13 at 4 34 30 pm" src="https://github.com/henninghall/react-native-date-picker/assets/381585/c3776901-0ca3-426f-9dc3-82684c6dfe2a">

**Comment by Nikhil-Rattan** [comment]
Same issue with the maximumDate

**Comment by usamabinishaq** [comment]
Same Issue

**Comment by usamabinishaq** [comment]
[soemarko](https://github.com/soemarko) did you find any solution?

**Comment by usamabinishaq** [comment]
> > [soemarko](https://github.com/soemarko) did you find any solution?
> 
> > If I commented out minimumDate, it works. I think this has something to do with the latest update.
> 
> That, for now, while I continue working on other things. I have some ideas for workaround, but worst case, just downgrade.

Yes For me I Downgraded the version to 4.3.5 and it worked

**Comment by henninghall** [comment]
I have publish an attempt to a temporary fix in version `4.3.7-alpha.0`, can someone with the error test it out?

**Comment by bishoymakkar** [comment]
if i back to minmum date it didnt excute on change

**Comment by henninghall** [comment]
Ok, thanks. When on 4.3.6: 
1. Are you using hermes? 
2. Is there a difference between having debugger on or off?

**Comment by bishoymakkar** [comment]
no difference in debugger 
and not using hermes

**Comment by soemarko** [comment]
> I have publish an attempt to a temporary fix in version `4.3.7-alpha.0`, can someone with the error test it out?
> 

Confirm this fixes it for me. Feel free to close this, but I think we should wait until official release?

**Comment by luihugo** [comment]
experiencing same error

**Comment by bishoymakkar** [comment]
`      <DatePicker
        style={styles.picker}
        date={date}
        onDateChange={this.handleOnDateChange}
        mode="date"
        locale={isArabic ? 'ar' : 'en'}
        textColor={colors.NAVY_HIGHEST}
        androidVariant="iosClone"
        minimumDate={minDate}
        maximumDate={maxDate}
      />`  when i used minimumDate and maximumDate while changing from date to minimum date it's didn't execute onDateChange function for android but it works good for ios

**Comment by henninghall** [comment]
What values have maxDate and minDate variables?

**Comment by bishoymakkar** [comment]
maximum date `new Date(moment().add(15, 'days'));`
minimum date `new Date();`

**Comment by CostasCF** [comment]
Reverting to v4.3.4 is an option for now till this is fixed.

**Comment by henninghall** [comment]
A fix is now released in version 4.3.7 (a better one than 4.3.7-alpha.0), give it a shot!

**Comment by 7dp** [comment]
Hi, experiencing the same issue.
But, I can confirm that 4.3.7 fixes it.

Really appreciate your HARD work on this dope package for the vast community! 🙌 @henninghall
